drivers/SmartThings/matter-appliance/src/matter-refrigerator/init.lua

```diff
@@ -135,17 +161,17 @@ local function setpoint_limit_handler(limit_field)
 end
 
 local function selected_temperature_level_attr_handler(driver, device, ib, response)
-  local tl_eps = embedded_cluster_utils.get_endpoints(device,clusters.TemperatureControl.ID, {feature_bitmap = clusters.TemperatureControl.types.Feature.TEMPERATURE_LEVEL})
-  if #tl_eps == 0 then
-    device.log.warn_with({ hub_logs = true }, string.format("Device does not support TEMPERATURE_LEVEL feature"))
+  if not supports_temperature_level_endpoint(device, ib.endpoint_id) then
```

Comment:
Similar to the comment above, I don't think we would need to check if the endpoint supports temperatureLevel because this handler would only be called for endpoints that send us an interaction for the `SelectedTemperatureLevel` handler:
```
    attr = {
      [clusters.TemperatureControl.ID] = {
        [clusters.TemperatureControl.attributes.TemperatureSetpoint.ID] = temperature_setpoint_attr_handler,
        [clusters.TemperatureControl.attributes.MinTemperature.ID] = setpoint_limit_handler(setpoint_limit_device_field.MIN_TEMP),
        [clusters.TemperatureControl.attributes.MaxTemperature.ID] = setpoint_limit_handler(setpoint_limit_device_field.MAX_TEMP),
        [clusters.TemperatureControl.attributes.SelectedTemperatureLevel.ID] = selected_temperature_level_attr_handler, -- <- here
```

Reply:
I'm sorry for the late reply.
I tested through the Virtual device app and confirmed that the `Temperature Setpoint` attribute is called even if `Temperature Control` does not support `TN` feature or `Supported Temperature Levels` or `Selected Temperature Levels` are called even if `TL` feature is not supported. So I added this code.